purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1569,10 +1577,8 @@ class Purchases internal constructor(
         listener: PurchaseErrorCallback
     ) {
         if (purchasingData.productType != ProductType.SUBS) {
-            dispatch {
-                listener.onError(PurchasesError(PurchasesErrorCode.PurchaseNotAllowedError,
-                    PurchaseStrings.UPGRADING_INVALID_TYPE).also { errorLog(it) }, false)
-            }
+            listener.dispatch(PurchasesError(PurchasesErrorCode.PurchaseNotAllowedError,
```

Comment:
does it make sense to add `getAndClearProductChangeCallback` here too? I think if the code enters here, the callback will not be cleared

Reply:
ohhh weird we do this check twice....once in the method that calls this one. though i suppose someone could independently call `replaceOldPurchaseWithNewProduct` too.. so maybe it's good to have both.

will add the clearproductchangecallback, good catch